src/errno.rs

```diff
@@ -3526,3 +3589,271 @@ mod consts {
+        EBADR = libc::EBADR,
+        EXFULL = libc::EXFULL,
+        ENOANO = libc::ENOANO,
+        EBADRQC = libc::EBADRQC,
+        EBADSLT = libc::EBADSLT,
+        EDEADLOCK = libc::EDEADLOCK,
+        EBFONT = libc::EBFONT,
+        ENOSTR = libc::ENOSTR,
+        ENODATA = libc::ENODATA,
+        ETIME = libc::ETIME,
+        ENOSR = libc::ENOSR,
+        ENONET = libc::ENONET,
+        ENOPKG = libc::ENOPKG,
+        EREMOTE = libc::EREMOTE,
+        ENOLINK = libc::ENOLINK,
+        EADV = libc::EADV,
+        ESRMNT = libc::ESRMNT,
+        ECOMM = libc::ECOMM,
+        EPROTO = libc::EPROTO,
+        EMULTIHOP = libc::EMULTIHOP,
+        EDOTDOT = libc::EDOTDOT,
+        EBADMSG = libc::EBADMSG,
+        EFTYPE = libc::EFTYPE,
+        ENOTUNIQ = libc::ENOTUNIQ,
+        EBADFD = libc::EBADFD,
+        EREMCHG = libc::EREMCHG,
+        ELIBACC = libc::ELIBACC,
+        ELIBBAD = libc::ELIBBAD,
+        ELIBSCN = libc::ELIBSCN,
+        ELIBMAX = libc::ELIBMAX,
+        ELIBEXEC = libc::ELIBEXEC,
+        ENOSYS = libc::ENOSYS,
+        ENOTEMPTY = libc::ENOTEMPTY,
+        ENAMETOOLONG = libc::ENAMETOOLONG,
+        ELOOP = libc::ELOOP,
+        EOPNOTSUPP = libc::EOPNOTSUPP,
+        EPFNOSUPPORT = libc::EPFNOSUPPORT,
+        ECONNRESET = libc::ECONNRESET,
+        ENOBUFS = libc::ENOBUFS,
+        EAFNOSUPPORT = libc::EAFNOSUPPORT,
+        EPROTOTYPE = libc::EPROTOTYPE,
+        ENOTSOCK = libc::ENOTSOCK,
+        ENOPROTOOPT = libc::ENOPROTOOPT,
+        ESHUTDOWN = libc::ESHUTDOWN,
+        ECONNREFUSED = libc::ECONNREFUSED,
+        EADDRINUSE = libc::EADDRINUSE,
+        ECONNABORTED = libc::ECONNABORTED,
+        ENETUNREACH = libc::ENETUNREACH,
+        ENETDOWN = libc::ENETDOWN,
+        ETIMEDOUT = libc::ETIMEDOUT,
+        EHOSTDOWN = libc::EHOSTDOWN,
+        EHOSTUNREACH = libc::EHOSTUNREACH,
+        EINPROGRESS = libc::EINPROGRESS,
+        EALREADY = libc::EALREADY,
+        EDESTADDRREQ = libc::EDESTADDRREQ,
+        EMSGSIZE = libc::EMSGSIZE,
+        EPROTONOSUPPORT = libc::EPROTONOSUPPORT,
+        ESOCKTNOSUPPORT = libc::ESOCKTNOSUPPORT,
+        EADDRNOTAVAIL = libc::EADDRNOTAVAIL,
+        ENETRESET = libc::ENETRESET,
+        EISCONN = libc::EISCONN,
+        ENOTCONN = libc::ENOTCONN,
+        ETOOMANYREFS = libc::ETOOMANYREFS,
+        EPROCLIM = libc::EPROCLIM,
+        EUSERS = libc::EUSERS,
+        EDQUOT = libc::EDQUOT,
+        ESTALE = libc::ESTALE,
+        ENOTSUP = libc::ENOTSUP,
+        ENOMEDIUM = libc::ENOMEDIUM,
+        EILSEQ = libc::EILSEQ,
+        EOVERFLOW = libc::EOVERFLOW,
+        ECANCELED = libc::ECANCELED,
+        ENOTRECOVERABLE = libc::ENOTRECOVERABLE,
+        EOWNERDEAD = libc::EOWNERDEAD,
+        ESTRPIPE = libc::ESTRPIPE,
+    }
+
+    impl Errno {
+        pub const EWOULDBLOCK: Errno = Errno::EAGAIN;
+        pub const EDEADLOCK: Errno = Errno::EDEADLK;
+        pub const EOPNOTSUPP: Errno = Errno::ENOTSUP;
+    }
+
+    pub const fn from_i32(e: i32) -> Errno {
```

Comment:
This function, on other platforms, [was deprecated](https://github.com/nix-rust/nix/blob/e4895d34d9e92b6d5693a7959bd5ef8bc98b53ba/src/errno.rs#L2110-L2115). On Cygwin, maybe we can simply make it private?